File: algo/code/04/c/05traverse.c
/* 遍历数组 */

#include <stdio.h>

int main() {
    int nums[4] = {1, 2, 3, 4};
    int length = sizeof(nums) / sizeof(nums[0]);
    int i = 0;

    // 通过索引遍历数组 while循环
    int sum1 = 0;
    while (i < length) {
        sum1 += nums[i];
        i++;
    }
    printf("%d\n", sum1); // 10

    // 通过索引遍历数组 for循环
    int sum2 = 0;
    for (i = 0; i < length; i++) {
        sum2 += nums[i];
    }
    printf("%d\n", sum2); // 10
    return 0;
}

File: algo/code/04/js/05traverse.js
/* 遍历数组 */

nums = [1, 2, 3, 4];

// 通过索引遍历数组 for循环
let sum1 = 0;
for (let i = 0; i < nums.length; i++) {
    sum1 += nums[i];
}
console.log(sum1); // 10

// 通过索引遍历数组 while循环
let sum3 = 0;
let i = 0;
while (i < nums.length) {
    sum3 += nums[i];
    i++;
}
console.log(sum3); // 10

// 直接遍历数组 for-of循环
let sum2 = 0;
for (const num of nums) {
    sum2 += num;
}
console.log(sum2); // 10